addons/actions/src/preview/action.ts

```diff
@@ -0,0 +1,25 @@
+import uuid from 'uuid/v1';
+import { addons } from '@storybook/addons';
+import { EVENT_ID } from '../constants';
+import { ActionDisplay, ActionOptions, HandlerFunction } from '../models';
+
+export function action(name: string, options: ActionOptions = {}): HandlerFunction {
+  const actionOptions = {
+    ...options,
+  };
+
+  // tslint:disable-next-line:no-shadowed-variable
+  const handler = function action(...args: any[]) {
+    const channel = addons.getChannel();
+    const id = uuid();
+    const actionDisplayToEmit: ActionDisplay = {
+      id,
+      count: 0,
+      data: { name, args },
```

Comment:
@gaetanmaisse Regarding my comment at https://github.com/storybooks/storybook/pull/5459#discussion_r255302247, you're constructing `ActionDisplay` here, which spreads the incoming function arguments into `args`, which you've also typed as `any[]`. Assuming this is the function where every storybook action call flows through, I think it's safe to also type `args` as `any[]` in `ActionDisplay`.

Reply:
Thanks! I missed that, type updated :)